### ISSUE TITLE
no-unit for zero number handles deg types

### ISSUE THREAD

**Issue by AndreyUtka** [stylelint/stylelint#1189]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Throw Unexpected unit for value with `deg`
Maybe It's bug. In last version of 5.2.1 this does not occur. 
> Which rule, if any, is this issue related to?

 `number-zero-length-no-unit`

> What CSS is needed to reproduce this issue?

```scss
.pr-button {
    background: linear-gradient(0deg, rgba(7, 29, 73, 0.12) 0%, rgba(7, 29, 73, 0) 80%) #fff;
}
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
        "number-leading-zero": "always",
        "number-no-trailing-zeros": true,
        "number-zero-length-no-unit": true,
  }
}
```

> Which version of stylelint are you using?

`6.2.2`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

https://github.com/vieron/stylelint-webpack-plugin

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes, it's related to SCSS maps

> What did you expect to happen?

"No warnings to be flagged."

> What actually happened (e.g. what warnings or errors you are getting)?

```shell
Unexpected unit on zero length number   number-zero-length-no-unit
```

**Comment by jeddy3** [comment]
@AndreyUtka Thanks for the accurate report.

Yes, this looks to be a regression. We need to [revert back](https://github.com/stylelint/stylelint/commit/db6bef6fde4160be779b956bf598f59e5e9a6b6b) to using a whitelist of known *length* units.

**Comment by AndreyUtka** [comment]
@jeddy3 great - thanks.

**Comment by evilebottnawi** [comment]
@jeddy3 tomorow i PR this issue

**Comment by jeddy3** [comment]
@evilebottnawi Thanks! You're on fire at the moment :)

@davidtheclark Perhaps this is an opportunity to get started on the on the [list initiative](https://github.com/stylelint/stylelint/issues/1071)? i.e. we create a `isKnownLengthUnit` here, which `isKnownUnit` will consume?

If it's needed down the line, we can make these two even more [granular](https://drafts.csswg.org/css-values/#lengths) e.g. `isKnownAbsoluteLengthUnit` and `isKnownFontRelativeUnit` etc.

@evilebottnawi It looks like the `q` unit is missing from that original whitelist, and so lets be careful not to lose it in the shuffle :)

**Comment by davidtheclark** [comment]
Or `isKnownUnit` could accept an options parameter to specify, e.g. `isKnownUnit(x, { only: "length" })`.

**Comment by jeddy3** [comment]
> Or isKnownUnit could accept an options parameter to specify, e.g. isKnownUnit(x, { only: "length" }).

Oh, that's much neater.

**Comment by evilebottnawi** [comment]
Close. Merged. Ref: https://github.com/stylelint/stylelint/pull/1196
